lib/util/RestConnection.js

```diff
@@ -112,6 +112,9 @@ module.exports = class RestConnection {
 
       case 'PUT':
         return this.post(fullUrl, params, additionalOptions, customHeaders, 'PUT');
+      
+      case 'HEAD':
+        return this.post(fullUrl, params, additionalOptions, customHeaders, 'HEAD');
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/HEAD

HEAD requests do not ever have a body, I think this should wrap around `this.get` instead of `this.post`.